### PR TITLE
Add initial Kubernetes generator support

### PR DESCRIPTION
**Pull request by saturnism**
WIP for #3443

### CODE REVIEW

File: generators/kubernetes/index.js [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+    prompting: {
+        askForPath: prompts.askForPath,
+
+        askForApps: prompts.askForApps,
+
+        askForClustersMode: prompts.askForClustersMode,
+
+        askForAdminPassword: prompts.askForAdminPassword,
+
+        askForKubernetesNamespace: function() {
```
Comment (deepu105) [inline generators/kubernetes/index.js]: Can we move all questions to promts file?

Comment (deepu105) [inline generators/kubernetes/index.js]: I think you missed the prompts.js file

**Comment by pascalgrimaud** [comment]
I wonder if we can use https://github.com/kubernetes/minikube to test this PR ?

**Comment by saturnism** [comment]
Yes! I would say give it a try.

**Comment by pascalgrimaud** [comment]
@saturnism : just made a PR to your branch -> fix eslint and add support for other databases. If you accept it, it should update this PR, right?

FYI, all my tests were with minikube, it rocks!

minikube

**Comment by pascalgrimaud** [comment]
I will start to write doc on this, unless someone else, who already uses Kubernetes, want to do it

**Comment by pkrishanth** [comment]
I am waiting for this.... When are you planning to release this ?

**Comment by pascalgrimaud** [comment]
As soon @saturnism merges my PR on his branch, then if someone can test this fresh PR, then we could merge this ! :)

**Comment by saturnism** [comment]
Done! Thanks so much @pascalgrimaud !

**Comment by ruddell** [comment]
I have the same result as @pascalgrimaud.  Mongo, postgresql, and mysql all work for me.  Great work!

If you choose an app using Cassandra, it fails silently on `Checking Docker images in applications' directories...`.  Could we add a message saying Cassandra is not supported (yet)?

**Comment by saturnism** [comment]
Actually we have a good way to deploy Cassandra now with the PetSets. Maybe
for the next version? :)

**Comment by pascalgrimaud** [comment]
Yes, Cassandra isnt support yet. Same for elasticsearch. It can be done later as it's in BETA

So this PR seems to be ready for beeing merged !

File: generators/kubernetes/prompts.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+var _ = require('lodash'),
+    dockerComposePrompts = require('../docker-compose/prompts');
+
+module.exports = _.extend({
```
Comment (deepu105) [inline generators/kubernetes/prompts.js]: @pascalgrimaud nice :clap:

**Comment by deepu105** [comment]
@pascalgrimaud yes please merge if you have tested it :+1:

**Comment by pascalgrimaud** [comment]
There are some improvements that can be done, but this 1st version is already ok! Thanks @saturnism !

**Comment by raulgomis** [comment]
Great work guys!! +1

**Pull request merged**